config.go

```diff
@@ -279,8 +279,8 @@ func (c *frozenConfig) getMaxByteSliceSize() int {
 
 func (c *frozenConfig) getMaxSliceAllocSize() int {
 	size := c.config.MaxSliceAllocSize
-	if size > maxAllocSize || size <= 0 {
-		return maxAllocSize
+	if int64(size) > maxAllocSize || size <= 0 {
+		return int(maxAllocSize)
```

Comment:
Hmm, if I change that to be an `int64` we end up with an issue in `codec_array.go` because `sliceType.UnsafeGrow` accepts an `int` and not an `int64`. Which means we'd just be pushing the overflow to that spot instead. 

Maybe we need to change the `maxAllocSize` to a different value for 32-bit systems?

Reply:
That seems reasonable. I am a little surprised the actual value is not different between these systems.

Reply:
I've updated this with my most recent commit. Let me know what you think